From 7883639e7c034f9e65d99ec106eb1e64c8544e92 Mon Sep 17 00:00:00 2001
From: Qusai Abudhaim <Qusai.Abudhaim@gmail.com>
Date: Wed, 22 Jul 2026 09:10:47 +0200
Subject: [PATCH 2/4] spi: reduce default SPI clock from 30 MHz to 10 MHz

Lower the default SPI clock frequency from 30 MHz to 10 MHz to improve
communication stability between the ESP firmware and the host.

The lower clock rate provides more reliable transfers on the current
hardware platform during SPI communication.

Higher clock rates caused packet dropping.
---
 .../esp_driver/network_adapter/main/spi_slave_api.c    | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/esp_hosted_ng/esp/esp_driver/network_adapter/main/spi_slave_api.c b/esp_hosted_ng/esp/esp_driver/network_adapter/main/spi_slave_api.c
index 9aadafd..6879a14 100644
--- a/esp_hosted_ng/esp/esp_driver/network_adapter/main/spi_slave_api.c
+++ b/esp_hosted_ng/esp/esp_driver/network_adapter/main/spi_slave_api.c
@@ -84,7 +84,7 @@ uint8_t g_spi_mode = SPI_MODE_3;
 #define GPIO_CS             10
 #define DMA_CHAN            ESP_SPI_CONTROLLER
 
-#define SPI_CLK_MHZ         30
+#define SPI_CLK_MHZ         10
 
 #elif defined CONFIG_IDF_TARGET_ESP32C3
 
@@ -95,7 +95,7 @@ uint8_t g_spi_mode = SPI_MODE_3;
 #define GPIO_CS             10
 #define DMA_CHAN            SPI_DMA_CH_AUTO
 
-#define SPI_CLK_MHZ         30
+#define SPI_CLK_MHZ         10
 
 #elif defined CONFIG_IDF_TARGET_ESP32S3
 
@@ -106,7 +106,7 @@ uint8_t g_spi_mode = SPI_MODE_3;
 #define GPIO_CS             10
 #define DMA_CHAN            SPI_DMA_CH_AUTO
 
-#define SPI_CLK_MHZ         30
+#define SPI_CLK_MHZ         10
 
 #elif defined CONFIG_IDF_TARGET_ESP32C2
 
@@ -117,7 +117,7 @@ uint8_t g_spi_mode = SPI_MODE_3;
 #define GPIO_CS             10
 #define DMA_CHAN            SPI_DMA_CH_AUTO
 
-#define SPI_CLK_MHZ         30
+#define SPI_CLK_MHZ         10
 
 #elif defined CONFIG_IDF_TARGET_ESP32C5
 
@@ -128,7 +128,7 @@ uint8_t g_spi_mode = SPI_MODE_3;
 #define GPIO_CS                10
 #define DMA_CHAN               SPI_DMA_CH_AUTO
 
-#define SPI_CLK_MHZ            30
+#define SPI_CLK_MHZ            10
 
 #elif defined CONFIG_IDF_TARGET_ESP32C6
 
-- 
2.43.0

